src/DynamoCoreWpf/ViewModels/Core/ConnectorViewModel.cs

```diff
@@ -504,7 +504,7 @@ public PreviewState PreviewState
                     return PreviewState.None;
                 }
 
-                if (Nodevm.ShowExecutionPreview)
+                if (Nodevm.ShowExecutionPreview || NodeEnd.ShowExecutionPreview)
```

Comment:
@QilongTang It seems to me like we should be watching both ends of the connector.
Can you confirm this ?

Reply:
Yes, good one!